src/cdi/CDI.hh

```diff
@@ -201,7 +201,7 @@ static double Planck2Rosseland(double const freq, double const exp_freq) {
   double factor;
 
   if (freq > 1.0e-5)
-    factor = NORM_FACTOR * (freq_3 * freq) / std::expm1(freq);
+    factor = NORM_FACTOR * (exp_freq * freq_3) * (freq / -std::expm1(-freq));
   else
```

Comment:
Since an optimizing compiler (above -O2) might remove the parentheses, let's add a DbC before factor is evaluated to ensure freq^4 doesn't cause overflow:
```
Check( freq < std::numeric_limits<double>::max()/freq_3 );
```
You might also need to `#include <limits>`

Since `freq>1e-5`, division by `std::expm1(-freq)` should be safe.